internal/pkg/agent/application/upgrade/artifact/download/fs/verifier.go

```diff
@@ -121,9 +122,14 @@ func (v *Verifier) verifyAsc(fullPath string, skipDefaultPgp bool, pgpSources ..
 			continue
 		}
 		raw, err := download.PgpBytesFromSource(check, v.client)
+		if errors.Is(err, download.ErrRemotePGPDownloadFailed) {
```

Comment:
Is there a way to avoid needing to duplicate this check into each individual verifier? 

Or at least is there a higher level test we can write that is guaranteed to try each possible download source so if we change this again in the future this can't be forgotten?

The integration test will catch the most important HTTP case at least.

Reply:
i moved check to `PgpBytesFromSource` used in both verifiers. this function filters these errors, logs warning and returns empty pgp. empty pgp is ignored as it was before.